### PR TITLE
Milestone 2: Creating Components

### PR DESCRIPTION
**Pull request by nganifaith**
Creating calculator user interface. This milestone makes use of react components to setup up the different layouts of my calculator app. We have four main components:

- Button component: Which creates out button layout
- ButtonPanel: Calls the Button component to create a collection of buttons
- Display: Handles the input and display of a computation
- App component: Which renders out ButtonPanel and display

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import Display from './Display';
+
+function App() {
+  return (
+    <div className="App">
+      <div className="calc-wrapper">
+        <Display />
+        <ButtonPanel />
+      </div>
+    </div>
```
Comment (bellom) [inline src/components/App.js]: You need to wrap both `Display` and `ButtonPanel` with [React.Fragment](https://reactjs.org/docs/fragments.html#short-syntax)
Reason:
We don't want to create an extra element, that's why you need to use fragment, so we are able to render without having to create an extra node wrapping `Display` and `ButtonPanel`.

Screenshot from Pathwright:
image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @nganifaith  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check my comments under this review;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best**

File: src/components/App.js [hunk 2]
```diff
@@ -5,10 +5,8 @@ import Display from './Display';
 function App() {
   return (
     <div className="App">
-      <div className="calc-wrapper">
-        <Display />
-        <ButtonPanel />
-      </div>
+      <Display />
+      <ButtonPanel />
     </div>
```
Comment (bellom) [inline src/components/App.js]: Oops, good job trying to make changes.

Let's fix the React Fragment together.

check out this example:
```
    <>
      <SomeComponent />
    </>
   ```

And if you really wanna style 

we could do this:
```
  <>
    <div>
      <SomeComponent />
    </div>
  </>
  ```

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @nganifaith   :clap: 

# Status: Changes Required! ♻️ 

**Kindly check out hint below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

File: src/components/App.js (hunk 2 above)
Comment (nganifaith) [reply src/components/App.js]: @bellom thank you for the review. Can you kindly check the code again. You comment is made rather on an old version

Comment (bellom) [reply src/components/App.js]: @nganifaith Oops, let's double-check, I think I commented on your new changes, as you can see up there, line 8-11 was replaced


@nganifaith I hope I was able to make everything more clear now.

Comment (nganifaith) [reply src/components/App.js]: @Bellom Thank you for the review, however is an unnecessary change. Which problem are you trying to solve with the fragment? I am using the class for styling. Please read the best practices. 
This isn't a use case for a Fragment I am calling two different components which are wrapped in one container. I understood in the first review I had two unnecessary containers. Look at the code again and tell me why I need a fragment in this case. And Please feel free to show the other to others

Comment (bellom) [reply src/components/App.js]: Apology, I think you can add style it is optional.

So if you stick to the style, we don't need the fragment as you said, sorry for the mixed up.

**Review by bellom — APPROVED** [review]
## Hi @nganifaith  :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Pull request merged**